src/VirtualClient/VirtualClient.Monitors/AmdSmiMetricQueryGpuParser.cs

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace VirtualClient.Monitors
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Data;
+    using System.Linq;
+    using System.Text.RegularExpressions;
+    using VirtualClient.Contracts;
+    using DataTableExtensions = VirtualClient.Contracts.DataTableExtensions;
+
+    /// <summary>
+    /// Parser for AmdSmi output document.
+    /// </summary>
+    public class AmdSmiMetricQueryGpuParser : MetricsParser
+    {
+        /// <summary>
+        /// Constructor for <see cref="AmdSmiMetricQueryGpuParser"/>
+        /// </summary>
+        /// <param name="rawText">Raw text to parse.</param>
+        public AmdSmiMetricQueryGpuParser(string rawText)
+            : base(rawText)
+        {
+        }
+
+        /// <inheritdoc/>
+        public override IList<Metric> Parse()
+        {
+            this.Preprocess();
+
+            List<Metric> metrics = new List<Metric>();
+            DataTable dataTable = DataTableExtensions.DataTableFromCsv(this.PreprocessedText);
+
+            foreach (DataRow row in dataTable.Rows)
+            {
+                Dictionary<string, IConvertible> metadata = new Dictionary<string, IConvertible>()
+                {
+                    { "gpu.id", Convert.ToString(SafeGet(row, "gpu")) }
+                };
+
+                metrics.Add(new Metric("utilization.gpu", Convert.ToDouble(SafeGet(row, "gfx_activity")), unit: "%", metadata: metadata));
+                double value = 100 * Convert.ToDouble(SafeGet(row, "used_vram")) / Convert.ToDouble(SafeGet(row, "total_vram"));
```

Comment:
This is not scalable, please see if you can generalize these lines

Reply:
Metric names in output of AMD SMI vary for every GPU architecture hence I am parsing as per requirement.